packages/editor/src/components/inner-blocks/index.js

```diff
@@ -144,20 +144,14 @@ InnerBlocks = compose( [
 	} ),
 	withDispatch( ( dispatch, ownProps ) => {
 		const {
-			replaceBlocks,
-			insertBlocks,
+			replaceInnerBlocks,
 			updateBlockListSettings,
 		} = dispatch( 'core/block-editor' );
 		const { block, clientId, templateInsertUpdatesSelection = true } = ownProps;
 
 		return {
 			replaceInnerBlocks( blocks ) {
-				const clientIds = map( block.innerBlocks, 'clientId' );
-				if ( clientIds.length ) {
-					replaceBlocks( clientIds, blocks );
-				} else {
-					insertBlocks( blocks, undefined, clientId, templateInsertUpdatesSelection );
-				}
+				replaceInnerBlocks( clientId, blocks, block.innerBlocks.length === 0 && templateInsertUpdatesSelection );
```

Comment:
`block.innerBlocks.length === 0` - can be computed inside `replaceInnerBlocks`. Is there any reason it has to be placed here?

Reply:
The selection is managed on a reducer that does not have access to the blocks themselves. In order to the check in a reducer, we would need a top-level high order reducer. We don't have any top-level higher-order reducer yet. I'm not sure we want to open that precedent.
But the main reason was to make easier to debug state changes. Currently, if the replaceInnerFlags action updateSelection flag is true the selection is updated if it is false the selection is not updated. If this logic was inside the reducer if the flag was false the selection was not updated, but if the flag was true the selection may or may not be updated.